app/models/manageiq/providers/redhat/inventory/parser/infra_manager.rb

```diff
@@ -498,7 +499,7 @@ def vm_hardware_guest_devices(persister_hardware, vm, addresses)
       network_id = vnic_profile.dig(:network, :id) if vnic_profile
       lan = if network_id
               switch = persister.host_virtual_switches.lazy_find(
-                :host    => persister_hardware.vm_or_template&.host,
+                :host    => host,
```

Comment:
@agrare so what you're saying is that `host_ems_ref` is blank, because the host is not assigned at this point?

So the old code should be ok, we just need to check here if `persister_hardware.vm_or_template&.host` is nil and not search for the switch here if it's nil.

Reply:
Yeah that's my guess at this point

Reply:
ok, updated the condition here